packages/eas-cli/src/build/build.ts

```diff
@@ -161,38 +171,33 @@ export async function prepareBuildRequestForPlatformAsync<
   };
 }
 
-const SERVER_SIDE_DEFINED_ERRORS = [
-  'TURTLE_DEPRECATED_JOB_FORMAT',
-  'EAS_BUILD_FREE_TIER_DISABLED',
-  'EAS_BUILD_FREE_TIER_DISABLED_IOS',
-  'EAS_BUILD_FREE_TIER_DISABLED_ANDROID',
-  'VALIDATION_ERROR',
-];
+const SERVER_SIDE_DEFINED_ERRORS: { [key: string]: any } = {
+  TURTLE_DEPRECATED_JOB_FORMAT: TurtleDeprecatedJobFormatError,
+  EAS_BUILD_FREE_TIER_DISABLED: EasBuildFreeTierDisabledError,
+  EAS_BUILD_FREE_TIER_DISABLED_IOS: EasBuildFreeTierDisabledIOSError,
+  EAS_BUILD_FREE_TIER_DISABLED_ANDROID: EasBuildFreeTierDisabledAndroidError,
+  VALIDATION_ERROR: RequestValidationError,
+};
 
 function handleBuildRequestError(error: any, platform: Platform): never {
   Log.debug(JSON.stringify(error.graphQLErrors, null, 2));
 
-  if (SERVER_SIDE_DEFINED_ERRORS.includes(error?.graphQLErrors?.[0]?.extensions?.errorCode)) {
-    Log.error(error?.graphQLErrors?.[0]?.message);
-    throw new Error('Build request failed.');
-  } else if (
-    error?.graphQLErrors?.[0]?.extensions?.errorCode === 'EAS_BUILD_DOWN_FOR_MAINTENANCE'
-  ) {
-    Log.error(
+  const graphQLErrorCode: string = error?.graphQLErrors?.[0]?.extensions?.errorCode;
+  if (Object.keys(SERVER_SIDE_DEFINED_ERRORS).includes(graphQLErrorCode)) {
```

Comment:
If TS won't complain about it then this is better:
```suggestion
  if (graphQLErrorCode in SERVER_SIDE_DEFINED_ERRORS) {
```